src/SerializingLink.ts

```diff
@@ -34,11 +34,11 @@ export default class SerializingLink extends ApolloLink {
         });
     }
     // Remove the first element from the queue and start the next operation
-    private dequeue = (key: string) => {
+    private dequeue = (key: string, observer: Observer<FetchResult>) => {
         if (!this.opQueues[key]) {
             return;
         }
-        this.opQueues[key].shift();
+        this.opQueues[key] = this.opQueues[key].filter(op => op.observer !== observer);
```

Comment:
Isn't `op.observer !== observer` always true when comparing two objects ? I believe this is stated here : https://basarat.gitbooks.io/typescript/docs/javascript/equality.html . Maybe you could use something like [deep-equal](https://www.npmjs.com/package/deep-equal) as they recommend.